pkg/controller/persistentvolume/persistentvolume_recycler_controller.go

```diff
@@ -132,23 +132,27 @@ func (recycler *PersistentVolumeRecycler) handleRecycle(pv *api.PersistentVolume
 	spec := volume.NewSpecFromPersistentVolume(pv, false)
 	plugin, err := recycler.pluginMgr.FindRecyclablePluginBySpec(spec)
 	if err != nil {
-		return fmt.Errorf("Could not find recyclable volume plugin for spec: %+v", err)
-	}
-	volRecycler, err := plugin.NewRecycler(spec)
-	if err != nil {
-		return fmt.Errorf("Could not obtain Recycler for spec: %+v", err)
-	}
-	// blocks until completion
-	err = volRecycler.Recycle()
-	if err != nil {
-		glog.Errorf("PersistentVolume[%s] failed recycling: %+v", pv.Name, err)
-		pv.Status.Message = fmt.Sprintf("Recycling error: %s", err)
 		nextPhase = api.VolumeFailed
-	} else {
-		glog.V(5).Infof("PersistentVolume[%s] successfully recycled\n", pv.Name)
-		nextPhase = api.VolumePending
+		pv.Status.Message = fmt.Sprintf("%v", err)
+	}
+
+	if plugin != nil {
+		volRecycler, err := plugin.NewRecycler(spec)
+		if err != nil {
+			return fmt.Errorf("Could not obtain Recycler for spec: %+v", err)
+		}
+		// blocks until completion
+		err = volRecycler.Recycle()
 		if err != nil {
-			glog.Errorf("Error updating pv.Status: %+v", err)
+			glog.Errorf("PersistentVolume[%s] failed recycling: %+v", pv.Name, err)
+			pv.Status.Message = fmt.Sprintf("Recycling error: %s", err)
+			nextPhase = api.VolumeFailed
```

Comment:
How do you suggest counting the number failures before giving up as a non-transient error?  Keep an internal map with volumeName/count and only set to Failed if count > x ?